spring-xd-dirt/src/main/java/org/springframework/xd/dirt/plugins/spark/streaming/SparkStreamingPlugin.java

```diff
@@ -100,20 +102,22 @@ public void postProcessModule(Module module) {
 			}
 			LocalMessageBusHolder messageBusHolder = new LocalMessageBusHolder();
 			messageBusHolder.set(module.getComponent(MessageBus.class));
-			receiver = new MessageBusReceiver(messageBusHolder, storageLevel, messageBusProperties, inboundModuleProperties);
+			receiver = new MessageBusReceiver(messageBusHolder, storageLevel, messageBusProperties,
+					inboundModuleProperties, getMimeType(module, true));
```

Comment:
I agree with David, and I think it might be worth 2 separate methods: getInputMimeType and getOutputMimeType, even if they in turn call a common private method to do part/most of the work.